applications/SeismicApplication/custom_conditions/fiber_beam_column_uniaxial_fiber.hpp

```diff
@@ -0,0 +1,283 @@
+     * @param NewId: id of the fiber
+     * @param Y: y coordinate
+     * @param Z: z coordinate
+     * @param pMaterial: pointer to the constitutive law
+     * @param pProperties: pointer to the properties
+     */
+    FiberBeamColumnUniaxialFiber(
+        IndexType NewId,
+        double Y,
+        double Z,
+        double Area,
+        ConstitutiveLaw::Pointer pMaterial,
+        PropertiesType::Pointer pProperties
+    );
+
+    /**
+     * Destructor
+     */
+    ~FiberBeamColumnUniaxialFiber() = default;
+
+    /**
+     * Copy Constructor
+     */
+    FiberBeamColumnUniaxialFiber(const FiberBeamColumnUniaxialFiber& rOther) = default;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    /**
+     * Assignement operator
+     */
+    FiberBeamColumnUniaxialFiber& operator=(const FiberBeamColumnUniaxialFiber& rOther) = default;
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /**
+     * Initializes the material law
+     */
+    void Initialize();
+
+    /**
+     * This method updates the strain in the fiber, calculates the material response,
+     * and stores the stress value.
+     * @param rSectionDeformationIncrements: the deformation increments of the section
+     */
+    void StateDetermination(const Vector& rSectionDeformationIncrements);
+
+    /**
+     * returns the global stiffness matrix of the fiber.
+     * @param rGlobalStiffnessMatrix
+     */
+    void CreateGlobalFiberStiffnessMatrix(Matrix& rGlobalStiffnessMatrix);
+
+    /**
+     * returns the global internal forces of the fiber.
+     * @param rGlobalStiffnessMatrix
+     */
+    void CreateGlobalFiberInternalForces(Vector& rGlobalFiberInternalForces);
+
+    /**
+     * Called when the non linear iterations have converged
+     */
+    void FinalizeSolutionStep();
+
+    ///@}
+    ///@name Access
+    ///@{
+
+    double const& GetStress() const {return mStress[0];}
+    double const& GetStrain() const {return mStrain[0];}
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const;
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const;
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const;
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+    ///@}
+
+
+protected:
+
+    ///@name Protected static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+    ///@}
+    ///@name Protected  Access
+    ///@{
+
+    ///@}
+    ///@name Protected Inquiry
+    ///@{
+
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+
+    ///@}
+
+
+private:
+
+    ///@name Static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Member Variables
+    ///@{
+
+    IndexType mId;       // id of the fiber
```

Comment:
Can you create a base class with the common member variables, methods, etc...?

Reply:
All the methods of sections and fibers are quite different, so I think if I do it would be a class with no implementation. I can create it if you think it's better to have. So it could be named `ElementComponent`?

Reply:
Could be, and move it to a different folder, like custom_classes if it is not a condition